applications/graphql-lambda/main.tf

```diff
@@ -0,0 +1,123 @@
+# Experimental: This module builds a serverless GraphQL instance.
+
+variable "environment" {
+  description = "The environment for this application: development, qa, or production."
+}
+
+variable "name" {
+  description = "The application name."
+}
+
+variable "papertrail_destination" {
+  description = "The Papertrail log destination for this application."
+}
+
+data "aws_ssm_parameter" "contentful_gambit_space_id" {
+  # All environments of this application use the same space.
+  name = "/contentful/gambit/space-id"
+}
+
+data "aws_ssm_parameter" "contentful_api_key" {
+  name = "/${var.name}/contentful/gambit/content-api-key"
+}
+
+data "aws_ssm_parameter" "northstar_auth_secret" {
+  name = "/northstar/${var.environment}/clients/${var.name}"
+}
+
+data "aws_ssm_parameter" "gambit_username" {
+  name = "/gambit/${local.gambit_env}/username"
+}
+
+data "aws_ssm_parameter" "gambit_password" {
+  name = "/gambit/${local.gambit_env}/password"
+}
+
+data "aws_ssm_parameter" "apollo_engine_api_key" {
+  name = "/${var.name}-lambda/apollo/api-key"
+}
+
+resource "random_string" "app_secret" {
+  length = 32
+}
+
+locals {
+  # Environment (e.g. 'dev' or 'DEV')
+  env = "${var.environment == "development" ? "dev" : var.environment}"
```

Comment:
![368635-20](https://user-images.githubusercontent.com/4562565/51919384-4d9f9b80-23b1-11e9-8f45-f9cfb261f22a.jpg)

Why might we have different `dev` vs `development`? Is this local vs Heroku hosted?

Reply:
The GraphQL app expects this environment variable to be provided as `dev`. We store environment as the full string `development` everywhere else in Terraform.

We should reconcile these so it's consistent everywhere!

Reply:
> Is this local vs Heroku hosted?

No, local apps would have `APP_ENV=local`. Our [development environment](http://docs.dosomething.org/engineering/overview#environments) is what developers connect their locals to when working on code (since we don't have a convenient way to spin up & maintain an entire local environment for folks, e.g. with Docker Compose)